File: allin-front-angular/src/app/subsystems/admin/models/commands/create-employee-command.ts
import { ExtendedFieldValueModel } from "../../../shared/models/extended-field-value-model";
import { CreatePersonCommand } from "./create-Person-command";

export interface CreateEmployeeCommand {
    personId?: number;
    departmentPositionId?: number;
    employeeCode?: string;
    employmentDate?: Date;
    contractTypeBaseId?: number;
    floating?: boolean;
    extendedFieldValues?: ExtendedFieldValueModel[];
}

File: Allin.Backend.Net/Microservices/Admin/Allin.Admin.Application/Commands/Employees/Create/CreateEmployeeCommand.cs
using Allin.Admin.Infrastructure.Persistence;
using Allin.Common.Utilities.Mappings;
using MediatR;

namespace Allin.Admin.Application.Commands
{
    public class CreateEmployeeCommand : IRequest<bool>, IMapFrom<Employee, CreateEmployeeCommand>
    {
        public long PersonId { get; set; }
        public long? DepartmentPositionId { get; set; }
        public string EmployeeCode { get; set; }
        public DateTime? EmploymentDate { get; set; }
        public long ContractTypeBaseId { get; set; }
        public bool Floating { get; set; }
        public IEnumerable<ExtendedFieldArg> ExtendedFieldValues { get; set; }
    }

}